### PR TITLE
Added support for reading ReadConcernLevel

### PR DESCRIPTION
**Pull request by slnowak**

### CODE REVIEW

**Comment by slnowak** [comment]
Do we support 3.6? If so, maybe there should be a support for afterClusterTime (https://docs.mongodb.com/manual/reference/read-concern/)

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java [hunk 1]
```diff
@@ -100,4 +92,8 @@ public MongoClientSettings settings() {
   public String database() {
     return database;
   }
+
+  private Optional<ReadConcern> maybeReadConcern(ConnectionString connectionString, JsonObject config) {
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java]: Maybe rename this to `setReadConcernIfExists` to be clearer.  

Also see Brian Goetz's guidance on using Optional in place of null checks https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555

File: vertx-mongo-client/pom.xml [hunk 2]
```diff
@@ -79,4 +79,13 @@
     </plugins>
   </build>
 
+  <dependencies>
```
Comment (karianna) [inline vertx-mongo-client/pom.xml]: Why not use the JUnit 4 parameterised runner?

File: vertx-mongo-client/src/test/java/io/vertx/ext/mongo/impl/config/ParsingReadConcernLevelTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(JUnitParamsRunner.class)
+public class ParsingReadConcernLevelTest {
```
Comment (karianna) [inline vertx-mongo-client/src/test/java/io/vertx/ext/mongo/impl/config/ParsingReadConcernLevelTest.java]: Really like the Given When Then style!

File: vertx-mongo-client/pom.xml (hunk 2 above)
Comment (slnowak) [reply vertx-mongo-client/pom.xml]: It looked more verbose. Also, I'm not sure if it's possible to parametrize over the method, not entire class. I can try to refactor that.

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java (hunk 1 above)
Comment (slnowak) [reply vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/MongoClientOptionsParser.java]: But it doesn't set anything, it just returns something that could be absent. What's your suggestion with that stackoverflow question exactly? It's not just a dummy getter.

File: vertx-mongo-client/pom.xml (hunk 2 above)
Comment (karianna) [reply vertx-mongo-client/pom.xml]: It's no biggy really - @johnoliver noted that this is slightly cleaner than the JUnit 4 way

**Pull request merged**